### PR TITLE
[PM-5540] `DesktopSettingsService`

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Migrates various desktop settings to use state providers. 

`getEnableAlwaysTop` was migrated, it was previously a user & global state value but we've migrated it to only being a global value. It's settable when no user is active, we are just discarding the user selected value in the migration because this way the tiebreaker in what a user had set will go to whichever account set it last.

A few unawaited promises were also fixed. 

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/desktop/src/main/messaging.main.ts [hunk 1]
```diff
@@ -17,19 +19,27 @@ export class MessagingMain {
...
+    private logService: LogService,
   ) {}
 
   init() {
     this.scheduleNextSync();
     if (process.platform === "linux") {
-      // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
-      // eslint-disable-next-line @typescript-eslint/no-floating-promises
-      this.stateService.setOpenAtLogin(fs.existsSync(this.linuxStartupFile()));
+      this.desktopSettingsService
```
Comment (MGibson1) [inline apps/desktop/src/main/messaging.main.ts:29]: ~~I'm confused -- why isn't this complaining about an unawaited promise?~~

the rule allows for `.catch` to dismiss the error, but I'm not sure this is really the same as awaiting or explicitly `void`ing. This feels less intentional promise handling to me.

Plus, doesn't it make sense to delay init until we've actually initialized the open at login setting?

File: apps/desktop/src/platform/services/desktop-settings.service.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  /**
+   * The application setting for whether or not the application should open at system login.
+   */
+  openAtLogin$ = this.openAtLoginState.state$.pipe(map((value) => value ?? false));
+
+  private readonly alwaysShowDockState = this.stateProvider.getGlobal(ALWAYS_SHOW_DOCK_KEY);
+  /**
+   * The application setting for whether or not the application should show up in the dock.
+   */
+  alwaysShowDock$ = this.alwaysShowDockState.state$.pipe(map((value) => value ?? false));
```
Comment (MGibson1) [inline apps/desktop/src/platform/services/desktop-settings.service.ts:79]: Shouldn't some of these defaults be platform dependent? Thinking particularly about show in dock, close to tray, and maybe others

**Review by MGibson1 — COMMENTED** [review]
I don't think I have anything here that needs changes, but I'm still confused as to where platform-specific defaults are set, which may be a new issue here or just something that isn't really handled right now.

**Review by MGibson1 — DISMISSED** [review]
Discussed offline, these settings are actually all consistent across platforms.

File: apps/desktop/src/main/messaging.main.ts (hunk 1 above)
Comment (justindbaur) [reply apps/desktop/src/main/messaging.main.ts]: Went with making this whole method `async` so I could just `await` these calls.

**Review by MGibson1 — APPROVED** [review]

**Review by MGibson1 — DISMISSED** [review]

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**